### ISSUE TITLE
[Elements] Client-side stripping of blinding key for blech32 addresses

### ISSUE THREAD

**Issue by shesek** [Blockstream/esplora#82]
Similarly to this, but for blech32 addresses:

https://github.com/Blockstream/esplora/blob/cad3cef99b1a149193e7f02bf515bbdc9851a91e/client/src/util.js#L39-L48

This requires a JavaScript implementation of the blech32 encoding format, which could be based on the [reference Python implementation](https://github.com/ElementsProject/elements/blob/master/test/functional/test_framework/liquid_addr.py) or the [Rust implementation in rust-elements](https://github.com/ElementsProject/rust-elements/pull/16/files#diff-02b0ed4def45fae054d60b6e1f8caf7c).

**Comment by stevenroose** [comment]
Rust WASM compile target will fix all of this. You want to prevent users from sending us their blinding pubkey?

**Comment by shesek** [comment]
Yes, that's the goal (and what's already happening with base58 encoded addresses). WASM could be a way to get this without having to port the blech32 implementation to JavaScript, but it'll come with a performance cost (primarily due to file size). But possibly worth it? @greenaddress thoughts?

**Comment by stevenroose** [comment]
I was mostly joking. I don't know if Rust and WASM is at a stable state or anything. It just seems like it would be awesome if we can have a web JS-compatible Bitcoin library based on rust-bitcoin. More like a dream, though..

**Comment by greenaddress** [comment]
I have libwally working in [wasm](https://github.com/greenaddress/libwally-core/commits/webassembly) (slightly experimental but I have used secp in wasm in emscripten for a while in the past).

https://github.com/ElementsProject/libwally-core/issues/97

I think wally results in smaller wasm for this/less deps? also since the rust lib depends on native secp anyway I am not sure if the rust to wasm is as easily available or requires some more tweaking.

**Comment by stevenroose** [comment]
I heard the WASM target for Rust is getting quite established. But I've never used it. It would be a nice experiment to get rust-bitcoin compiled to WASM and perhaps a JS FFI API being created.

**Comment by greenaddress** [comment]
Given wally is already done I plan to use that for this.
